### PR TITLE
Implement-login-design

### PR DESCRIPTION
**Pull request by rrarima**
Implement-login-design

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -5,15 +5,79 @@ const request = require('request');
...
+app.post('/login', async (req, res) => {
+  console.log('Login endpoint hit. Request body:', req.body)
+  const { username, password } = req.body;
+
+  const user = await User.findOne({ where: { username: username } });
+  if (!user) {
+    return res.status(400).json({ error: 'User not found' });
+  }
+
+  const validPassword = await bcrypt.compare(password, user.password);
```
Comment (Lestt) [inline server.js:82]: You are checking an encrypted version of the password, whereas on the subscription we never encrypt it in your database.
You may want to talk with the other developer about why you are doing this here.

File: server.js [hunk 2]
```diff
@@ -5,15 +5,79 @@ const request = require('request');
...
+  if (!validPassword) {
+    return res.status(400).json({ error: 'Invalid password' });
+  }
+  console.log('Login successful:', user);
+  res.json({
```
Comment (Lestt) [inline server.js:89]: Just as a POC and show how the login is working, you can return those sensitive data in clear. But as soon as you understand what you should return you will have to replace this.

File: server.js (hunk 1 above)
Comment (rrarima) [reply server.js:82]: I'll discuss this with the team. I had a few old user data that where encrypted originally so i had to use this to read them specifically.

File: server.js (hunk 2 above)
Comment (rrarima) [reply server.js:89]: I've added a token authentication library and removed the sensitive data in replacement of the token instead. Is this a good way?

File: server.js (hunk 1 above)
Comment (Lestt) [reply server.js:82]: I think encrypting your password in the database is a really good and safe practice, just wanted to be sure you understand why you need this, and realize this should be done at the registration as well.

File: server.js [hunk 3]
```diff
@@ -4,16 +4,93 @@ const { Sequelize, DataTypes } = require('sequelize');
...
+  if (!user) {
+    return res.status(400).json({ error: 'User not found' });
+  }
+
+  const validPassword = await bcrypt.compare(password, user.password);
+  if (!validPassword) {
+    return res.status(400).json({ error: 'Invalid password' });
+  }
+
+  const token = jwt.sign({ id: user.id, username: user.username }, config.JWT_KEY, { expiresIn: '1s' });
```
Comment (Lestt) [inline server.js]: I think you just need the user ID in the token.

File: server.js [hunk 4]
```diff
@@ -47,9 +64,33 @@ app.post('/registration', async (request, response) => {
...
+  if (!user) {
+    return res.status(400).json({ error: 'User not found' });
+  }
+
+  const validPassword = await bcrypt.compare(password, user.password);
+  if (!validPassword) {
+    return res.status(400).json({ error: 'Invalid password' });
+  }
+
+  const token = jwt.sign({ id: user.id }, config.JWT_KEY, { expiresIn: '1h' });
```
Comment (Lestt) [inline server.js:87]: I was about to ask and forgot that `1s` seemed a bit short for expiration time 😁
I suppose you figured it out by yourself.

Comment (rrarima) [reply server.js:87]: Ah yes I was trying to test if the token expiration was working 👍

File: server.js [hunk 5]
```diff
@@ -1,33 +1,55 @@
...
+//     req.user = user;
+//     next();
+//   });
+// }
 
 app.post('/registration', async (request, response) => {
   const { username, email, password } = request.body;
-  console.log(request.body);
-  if (username === null || username.length < 1)  {
+  console.log(username);
```
Comment (Lestt) [inline server.js]: Do you need this in production?

File: server.js [hunk 6]
```diff
@@ -47,9 +69,33 @@ app.post('/registration', async (request, response) => {
...
   } catch (error) {
     console.error(error);
-    response.status(400).json(error.errors[0].message);
-  } 
+    response.status(400).send(error.errors[0].message);
+  }
+});
+
+app.post('/login', async (req, res) => {
+  console.log('Login endpoint hit. Request body:', req.body)
```
Comment (Lestt) [inline server.js]: Same here

File: server.js [hunk 7]
```diff
@@ -47,9 +69,33 @@ app.post('/registration', async (request, response) => {
...
+  }
+
+  const validPassword = await bcrypt.compare(password, user.password);
+  if (!validPassword) {
+    return res.status(400).json({ error: 'Invalid password' });
+  }
+
+  const token = jwt.sign({ id: user.id }, config.JWT_KEY, { expiresIn: '1h' });
+
+  console.log('Login successful:', user);
```
Comment (Lestt) [inline server.js]: And last one.

**Pull request merged**